ProjFS.Mac/PrjFSKextTests/MockProc.cpp

```diff
@@ -22,3 +22,43 @@ proc_t vfs_context_proc(vfs_context_t ctx)
 {
     return NULL;
 }
+
+proc_t proc_self(void)
+{
+    return NULL;
+}
+
+kauth_cred_t kauth_cred_proc_ref(proc_t procp)
+{
+    return NULL;
+}
+
+uid_t kauth_cred_getuid(kauth_cred_t _cred)
+{
+    // Values over 500 are non-system processes
+    return 501;
+}
+
+void kauth_cred_unref(kauth_cred_t *_cred)
+{
+}
+
+int proc_ppid(proc_t)
+{
+    return 1;
+}
+
+proc_t proc_find(int pid)
+{
+    return NULL;
+}
+
+int proc_rele(proc_t p)
+{
+   return 1;
```

Comment:
Any number will actually due since we don't test the result in code.
The results here will seem a bit odd, but hopefully make a lot more sense when we unit tests CurrentProcessWasSpawnedByRegularUser.